addons/mail/models/mail_tracking_value.py

```diff
@@ -81,6 +81,11 @@ def create_tracking_values(self, initial_value, new_value, col_name, col_info, t
                 'old_value_char': initial_value and initial_value.sudo().name_get()[0][1] or '',
                 'new_value_char': new_value and new_value.sudo().name_get()[0][1] or ''
             })
+        elif col_info['type'] in ['many2many', 'one2many']:
+            values.update({
+                'old_value_char': initial_value and ', '.join(initial_value.mapped('display_name')) or '',
+                'new_value_char': new_value and ', '.join(new_value.mapped('display_name')) or ''
```

Comment:
changed `name` to `display_name` since `name` is not guaranteed to be in the model fields whereas `display_name` is magically managed by the ORM.